File: largest_sum_contiguous_subarray/largest_sum_contiguous_subarray.py
def largest_sum_contiguous_subarray(arr):
    """
    Find the subarray with maximum sum from all subarrays.

    :param arr: List of numbers to form subarray from.
    :return: The maximum sum in all subarrays.
    """
    max_now = 0
    max_next = 0
    for i in arr:
        max_next += i
        max_now = max(max_next, max_now)
        max_next = max(0, max_next)
    return max_now


def main():
    arr = [-2, -3, 4, -1, -2, 1, 5, -3]
    print('Maximum contiguous sum is', largest_sum_contiguous_subarray(arr))


if __name__ == '__main__':
    main()

File: largest_sum_contiguous_subarray/largestSumContiguousSubarray.js
function largestSumContiguousSubarray (arr) {
	/*
	* Find the subarray with maximum sum from all subarrays.
	* :param arr: List of numbers to form subarray from.
	* :return: The maximum sum in all subarrays.
	*/
	let maxPrev = 0;
	let maxNow = 0;
	for (let x = 0; x < arr.length; x++) {
		maxNow += arr[x];
		maxPrev = Math.max(maxPrev, maxNow);
		maxNow = Math.max(0, maxNow);
	}
	return maxPrev;
}

function main () {
	let arr = [-2, -3, 4, -1, -2, 1, 5, -3];
	console.log('Maximum contiguous sum is : ' + largestSumContiguousSubarray(arr));
}

main();
